### PR TITLE
check mine address

### PR DESCRIPTION
**Pull request by lvaccaro**
check if the input address from uri is mine or show an error: "Not mine address"

### CODE REVIEW

File: src/server.rs [hunk 1]
```diff
@@ -81,6 +81,16 @@ pub fn create_server(conf: ConfigOpts, wallet: Wallet<AnyBlockchain, Tree>) -> S
                     Some(address) => address,
                     None => return Ok(response.body(not_found().as_bytes().to_vec())?),
                 };
+                match is_mine_address(&*wallet, address) {
```
Comment (notmandatory) [inline src/server.rs:84]: small English language nit.. I think this and below should be "is_address_mine" or "is_my_address"

File: src/server.rs [hunk 2]
```diff
@@ -81,6 +81,16 @@ pub fn create_server(conf: ConfigOpts, wallet: Wallet<AnyBlockchain, Tree>) -> S
...
+                    Ok(mine) => {
+                        if !mine {
+                            return Ok(response.body(
+                                format!("Not mine address {}", address).as_bytes().to_vec(),
```
Comment (notmandatory) [inline src/server.rs:88]: same nit as above.. should read "Not my address: {}" or maybe "Address {} is not mine"

File: src/server.rs (hunk 1 above)
Comment (lvaccaro) [reply src/server.rs:84]: `is_my_address()` sounds great, thanks

File: src/server.rs (hunk 2 above)
Comment (lvaccaro) [reply src/server.rs]: thanks

File: src/server.rs [hunk 3]
```diff
@@ -120,6 +132,17 @@ fn last_unused_address(wallet: &Wallet<AnyBlockchain, Tree>) -> Result<Address,
     wallet.get_address(LastUnused)
 }
 
+fn is_my_address(
+    wallet: &Wallet<AnyBlockchain, Tree>,
+    addr: &str,
+) -> Result<bool, simple_server::Error> {
+    wallet.sync(log_progress(), None).map_err(|_| gen_err())?;
```
Comment (RCasatta) [inline src/server.rs:139]: `sync` is a heavy operation, we should check if the address `wallet.is_mine` first, and only if false call the sync and then the `wallet.is_mine` again.
This is still dos-able, but much better in the best/most common case

**Pull request merged**